smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaConnector.java

```diff
@@ -84,7 +83,7 @@
 @ConnectorAttribute(name = "partition", type = "int", direction = Direction.OUTGOING, description = "The target partition id. -1 to let the client determine the partition", defaultValue = "-1")
 @ConnectorAttribute(name = "waitForWriteCompletion", type = "boolean", direction = Direction.OUTGOING, description = "Whether the client waits for Kafka to acknowledge the written record before acknowledging the message", defaultValue = "true")
 @ConnectorAttribute(name = "max-inflight-messages", type = "long", direction = Direction.OUTGOING, description = "The maximum number of messages to be written to Kafka concurrently. It limits the number of messages waiting to be written and acknowledged by the broker. You can set this attribute to `0` remove the limit", defaultValue = "1024")
-@ConnectorAttribute(name = "cloud-events", type = "boolean", direction = Direction.OUTGOING, description = "Enables (default) or disables the Cloud Event support. If enabled, the connector sends the outgoing messages as Cloud Event if the message to be sent includes Cloud Event Metadata.", defaultValue = "true")
+@ConnectorAttribute(name = "cloud-events", type = "boolean", direction = Direction.INCOMING_AND_OUTGOING, description = "Enables (default) or disables the Cloud Event support. If enabled, the connector sends the outgoing messages as Cloud Event if the message to be sent includes Cloud Event Metadata and analyzes the incoming records and try to create Cloud Event metadata.", defaultValue = "true")
```

Comment:
It's probably a good idea to move the attribute definition to the top where the other Incoming and Outgoing attributes are defined

Reply:
And maybe a test should check that there are no duplicates? It shouldn't be too hard to implement.

Reply:
This is not a duplicate (duplicate would cause compilation errors):


We have:
```
@ConnectorAttribute(name = "cloud-events", type = "boolean", direction = Direction.INCOMING, description = "Enables (default) or disables the Cloud Event support. If enabled, the connector analyzes the incoming records and try to create Cloud Event metadata.", defaultValue = "true")


@ConnectorAttribute(name = "cloud-events", type = "boolean", direction = Direction.OUTGOING, description = "Enables (default) or disables the Cloud Event support. If enabled, the connector sends the outgoing messages as Cloud Event if the message to be sent includes Cloud Event Metadata.", defaultValue = "true")
```

The directions are different. The first one is for INCOMING, the second one is for OUTGOING. It's not the same attribute. They share the same name but are inherently different attributes.